### PR TITLE
[KYUUBI #2473] [Improvement] Add FilteredShowDatabasesCommand to AuthZ module

### PR DESCRIPTION
**Pull request by zhaomin1423**
### _How was this patch tested?_
- [x] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by yaooqinn** [comment]
can we run `build/mvn clean package -pl :kyuubi-spark-authz_2.12 -Dspark.version=2.4.7`? manually?

**Comment by yaooqinn** [comment]
> > can we run `build/mvn clean package -pl :kyuubi-spark-authz_2.12 -Dspark.version=2.4.7`? manually?
> 
> It can'e compiled using 2.4.7, the ShowNamespaces was added in Spark3

try to make it work? I guess it's easy

**Comment by packyan** [comment]
> > can we run `build/mvn clean package -pl :kyuubi-spark-authz_2.12 -Dspark.version=2.4.7`? manually?
> 
> It can'e compiled using 2.4.7, the ShowNamespaces was added in Spark3

Try using the `String` of node names in pattern matching and add `ShowDatabasesCommand` case to adapt spark 2.x.

**Comment by zhaomin1423** [comment]
Thanks, using reflecting make it work.

**Comment by yaooqinn** [comment]
can we handle ShowNamespace in PlannerStrategy?

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala [hunk 1]
```diff
@@ -55,6 +80,62 @@ case class FilteredShowTablesCommand(delegated: RunnableCommand)
     val result = SparkRangerAdminPlugin.isAccessAllowed(request)
     result != null && result.getIsAllowed
   }
+}
+
+case class FilteredShowNamespacesCommand(delegated: Any) extends RunnableCommand
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala:85]: delegated: SparkPlan

File: extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala [hunk 2]
```diff
@@ -55,6 +80,62 @@ case class FilteredShowTablesCommand(delegated: RunnableCommand)
...
+    output.asInstanceOf[Seq[Attribute]]
+  }
+
+  override def run(sparkSession: SparkSession): Seq[Row] = {
+    val runMethod = delegated.getClass.getMethod("run")
+    runMethod.setAccessible(true)
+    val rows = runMethod.invoke(delegated).asInstanceOf[Seq[InternalRow]]
+    val ugi = AuthZUtils.getAuthzUgi(sparkSession.sparkContext)
+    rows.filter(r => isAllowed(r, ugi)).map(r =>
+      Row.fromSeq(Seq.apply(r.getString(0).toString)))
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleReplaceShowObjectCommands.scala:101]: ~.toString~

File: extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala [hunk 3]
```diff
@@ -273,6 +273,17 @@ abstract class RangerSparkExtensionSuite extends KyuubiFunSuite with SparkSessio
       doAs("admin", sql(s"DROP DATABASE IF EXISTS $db"))
     }
   }
+
+  test("show databases") {
+    val db = "default2"
+    try {
+      doAs("admin", sql(s"CREATE DATABASE IF NOT EXISTS $db"))
+      doAs("admin", assert(sql(s"SHOW DATABASES").collect().length == 2))
+      doAs("bob", assert(sql(s"SHOW DATABASES").collect().length == 1))
```
Comment (yaooqinn) [inline extensions/spark/kyuubi-spark-authz/src/test/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RangerSparkExtensionSuite.scala:282]: match with value

**Review by yaooqinn — APPROVED** [review]

**Comment by yaooqinn** [comment]
thanks, merged to master

**Comment by zhaomin1423** [comment]
Thanks for your suggestions.

**Pull request closed without merging**